gdal_virtual_file_system.py

```diff
@@ -0,0 +1,27 @@
+"""
+Class that manages open and closing of a single gdal file.
+The pattern is to instantiate the class, call create to open it and then 
+call close to close the file.
+"""
+import os
+from osgeo import gdal
+
+class GDALVirtualFileSystem:
```

Comment:
General idea: Maybe instead of having the create and close methods you can do the following. Move all the logic of create into the init function and move the close function into the destructor. That way users just have to create the object and when it gets garbage collected the gdal stuff gets cleaned up. Might provide for a cleaner interface!

Reply:
hey I thought about something similar when I first implemented this but got stuck on how python handles destructors/clean up. I found a better solution though, using with statements: https://stackoverflow.com/questions/865115/how-do-i-correctly-clean-up-a-python-object. What do you think?

Reply:
I think that would provide a nice interface. Specially since it has a lot of parallels to how most people interact with file objects (which is through "with" statements).